cli/progress/multiprogress.go

```diff
@@ -222,63 +328,101 @@ func (m modelMultiProgress) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 	}
 }
 
-func (m modelMultiProgress) updateOverallProgress() tea.Cmd {
+func (m *modelMultiProgress) allDone() bool {
+	finished := 0
+	for k := range m.Progress {
+		if k == overallProgressIndexName {
+			continue
+		}
+		m.lock.Lock()
```

Comment:
this method could probably be simplified by locking and the top and defer unlocking right after it instead of locking and unlocking for each individual thing